optuna/storages/rdb/storage.py

```diff
@@ -912,61 +914,169 @@ def get_trial_number_from_id(self, trial_id):
     def get_trial(self, trial_id):
         # type: (int) -> FrozenTrial
 
-        return self._get_and_cache_trial(trial_id)
+        session = self.scoped_session()
 
-    def _get_and_cache_trial(self, trial_id, deepcopy=True):
-        # type: (int, bool) -> FrozenTrial
+        trial_model = (
+            session.query(
+                models.TrialModel.trial_id,
+                models.TrialModel.state,
+                models.TrialModel.value,
+                models.TrialModel.datetime_start,
+                models.TrialModel.datetime_complete,
+                models.TrialModel.number,
+                models.TrialModel.study_id,
+            )
+            .filter(models.TrialModel.trial_id == trial_id)
+            .one_or_none()
+        )
 
-        cached_trial = self._finished_trials_cache.get_cached_trial(trial_id)
-        if cached_trial is not None:
-            if deepcopy:
-                return copy.deepcopy(cached_trial)
-            else:
-                return cached_trial
+        if not trial_model:
+            raise KeyError("No trial with trial-id {} found.".format(trial_id))
 
-        session = self.scoped_session()
+        frozen_trial = self._get_trials_from_trial_models(session, [trial_model])[0]
 
-        trial = models.TrialModel.find_or_raise_by_id(trial_id, session)
-        params = models.TrialParamModel.where_trial(trial, session)
-        values = models.TrialValueModel.where_trial(trial, session)
-        user_attributes = models.TrialUserAttributeModel.where_trial(trial, session)
-        system_attributes = models.TrialSystemAttributeModel.where_trial(trial, session)
+        return copy.deepcopy(frozen_trial)
```

Comment:
Just out of curiosity, why is this deep copy needed here?

Reply:
It will be unnecessary after #1274 and I'll remove the deep-copy as #1274 will be merged soon.

Reply:
I see. Thank you for the explanation!